packages/hydrogen/src/pagination/Pagination.ts

```diff
@@ -158,9 +181,31 @@ export function usePagination<NodesType>(
   startCursor: Maybe<string> | undefined;
   endCursor: Maybe<string> | undefined;
 } {
-  const {state, search} = useLocation() as {
+  if (!connection.pageInfo) {
+    makeError('pageInfo');
+  }
+
+  if (typeof connection.pageInfo.startCursor === 'undefined') {
+    makeError('pageInfo.startCursor');
```

Comment:
Both our templates did _not_ have `pageInfo.startCursor`. This is required for the pagination component to work properly. Now we validate and make sure it's present.